setup_grammar.py

```diff
@@ -1,70 +1,128 @@
+import datetime
 import pathlib
 import subprocess
 from enum import Enum
 
 import requests
 
-GRAMMAR = {
-    'lexer': {
-        'url': 'https://raw.githubusercontent.com/crate/crate/master/libs/sql-parser/src/main/antlr/io/crate/sql/parser/antlr/SqlBaseLexer.g4',
-        'filename': 'SqlBaseLexer.g4'
+
+class Antlr4Target(Enum):
+    js = 'JavaScript'
+    python = 'Python3'
+
+
+build_options = {
+    'antlr4_compiled_target_output': {
+        Antlr4Target.js: 'cratedb_sqlparse_js',
+        Antlr4Target.python: 'cratedb_sqlparse_py'
     },
-    'parser': {
-        'url': 'https://raw.githubusercontent.com/crate/crate/master/libs/sql-parser/src/main/antlr/io/crate/sql/parser/antlr/SqlBaseParser.g4',
-        'filename': 'SqlBaseParser.g4'
-    }
 
+    'antlr4_compiled_target_subdir': 'parser/generated_parser',
+
+    # List of '.g4' files that will be built
+    'files': [
+        {
+            'url': 'https://raw.githubusercontent.com/crate/crate/{version}/libs/sql-parser/src/main/antlr/io/crate/sql/parser/antlr/SqlBaseLexer.g4',
```

Comment:
Makes a shorter URL, but adds a redirect at runtime. It's probably a matter of taste.

```suggestion
            'url': 'https://github.com/crate/crate/{version}/libs/sql-parser/src/raw/antlr/io/crate/sql/parser/antlr/SqlBaseLexer.g4',
```

Reply:
`https://github.com/crate/crate/5.6.4/libs/sql-parser/src/raw/antlr/io/crate/sql/parser/antlr/SqlBaseLexer.g4` doesn't seem to be valid

Reply:
That one is a valid link. I've conveyed the `raw` at the wrong place. Apologies.

https://github.com/crate/crate/raw/5.6.4/libs/sql-parser/src/main/antlr/io/crate/sql/parser/antlr/SqlBaseLexer.g4

Reply:
gotcha